src/controllers/transaction/index.ts

```diff
@@ -0,0 +1,56 @@
+import { Controller, Get, Route, Path, Response, Tags, Security, Query } from 'tsoa'
+import type { Logger } from 'pino'
+import { logger } from '../../lib/logger'
+import Database from '../../lib/db'
+import { UUID } from '../../models/uuid'
+import { BadRequest, NotFound } from '../../lib/error-handler/index'
+import { TransactionResponse, TransactionType } from '../../models/transaction'
+
+@Route('transaction')
+@Tags('transaction')
+@Security('bearerAuth')
+export class TransactionController extends Controller {
+  log: Logger
+  db: Database
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/transaction' })
+    this.db = new Database()
+  }
+
+  /**
+   * Returns the details of all transactions.
+   * @summary List all transactions
+   * @Query apiType lists all transactions by that type
+   */
+  @Response<BadRequest>(400, 'Request was invalid')
+  @Response<NotFound>(404, 'Item not found')
+  @Get('/')
+  public async getAllTransactions(@Query() apiType?: TransactionType): Promise<TransactionResponse[]> {
+    if (apiType) {
+      const transactionsByType = await this.db.getTransactionsByType(apiType)
+      if (!transactionsByType) throw new NotFound('transactionsByType')
```

Comment:
if you give an invalid type to `getTransactionsByType` it will find no matches in the db and return `[]`. That won't cause this `if` to evaluate to `true` though so the `NotFound` doesn't get thrown. I think you'll need to check the `apiType` query is a valid value for the enum instead and throw an error if it's not

Reply:
Yea. when you get a moment I think this might be resolved, will double check in a sec.

Reply:
I don't think the line `if (!transactionsByType) throw new NotFound('transactionsByType')` is needed? `transactionsByType` should never be falsy as the db query will return a `[]` if there's no match, and we can return `[]` rather than error